File: src/scala/org/pantsbuild/zinc/compiler/native-image-substitutions/Substitutions.java
package org.pantsbuild.zinc.compiler.substitutions.scala.tools.nsc;

// NB: This is necessary to avoid the javadoc task erroring out due to the lack of public classes.
public class Substitutions {}

final class Target_scala_reflect_internal_util_StatisticsStatics {
   static boolean areSomeHotStatsEnabled() {
     return false;
   }
   static boolean areSomeColdStatsEnabled() {
     return false;
   }

  public static void enableColdStats() {
    throw new RuntimeException("Operation unsupported enable cold stats in native scala compiler.");
  }

  public static void disableColdStats() {
    throw new RuntimeException(
      "Operation disable cold stats is unsupported in native scala compiler.");
  }

  public static void enableHotStats() {
    throw new RuntimeException(
      "Operation enable hot stats is unsupported in native scala compiler.");
  }

  public static void disableHotStats() {
    throw new RuntimeException(
      "Operation disable hot stats is unsupported in native scala compiler.");
  }
}

File: src/scala/org/pantsbuild/zinc/compiler/native-image-substitutions/substitutions.scala
package org.pantsbuild.zinc.compiler.substitutions.scala.tools.nsc

import java.io.File

final class Target_scala_tools_nsc_ast_TreeBrowsers$SwingBrowser {
  def browse(pName: String, units: List[AnyRef]): Unit = throw new RuntimeException("Swing currently unsupported in the native compiler.")
}

final class Target_sbt_io_IO$ {
  def getModifiedTimeOrZero(file: File): Long = file.lastModified()
  def setModifiedTimeOrFalse(file: File, mtime: Long): Boolean = file.setLastModified(mtime)
}
